docs/sidechain/spec.md

```diff
@@ -0,0 +1,945 @@
+   cross-chain and the reward amount will be transferred from the source account
+   to the door account. Collecting rewards is discussed below. This transaction
+   will create a `XChainTransferRewardState` ledger object for this transaction.
+   It will be owned by the door account.
+   
+3) When a witness servers sees a new cross-chain transaction, it submits a
+   transaction on the destination chain that adds a signature witnessing the
+   cross-chain transaction. This will include the amount being transferred
+   cross-chain, the reward amount, the account to send the reward to, and the
+   optional destination account. These signatures will be accumulated on the
+   cross-chain sequence number object. The keys used in these signatures must
+   match the keys on the multi-signers list on the door account.
+
+4) When a quorum of signatures have been collected, the cross-chain funds can be
+   claimed on the destination chain. If a destination account is specified, the
+   funds will move when the transaction that adds the last signature is executed
+   (the signature that made the quorum). On success, the cross-chain sequence
+   number object is removed. If there is an error (for example, the destination
+   has deposit auth set) or the optional destination account was not specified,
+   a "cross-chain claim" transaction must be use. Note, if the signers list on
+   the door account changes while the signatures are collected, the signers list
+   at the time the quorum is reached is controlling. When the quorum is reach,
+   the signing keys will be checked against the current signers list, and if a
+   collected signature's key is no longer on that list it is removed and
+   signatures will continue to be collected.
+   
+5) On the destination chain, the the owner of the "cross chain sequence number"
+   (see 1) can submit a "cross chain claim" transaction that includes the "cross
+   chain sequence number", the sidechain, and the destination. The "cross chain
+   sequence number" object must exist and must have already collected enough
+   signatures from the witness servers for this to succeed. On success, a
+   payment is made from the door account to the destination and the "cross chain
+   sequence number" is deleted. A "cross chain claim" transaction can only
+   succeed once, as the "cross chain sequence number" for that transaction can
+   only be created once. In case of error, the funds can be sent to an alternate
+   account and eventually returned to the initiating account. Note that this
+   transaction is only used if the optional destination account is not specified
+   in step (2) or there is an error when sending funds to that destination
+   account.
+
+6) When a witness server sees funds being claimed on the destination chain, it
+   submits a transaction on the source chain allowing the signature rewards to
+   be collected. It collects these signatures on the `XChainTransferRewardState`
```

Comment:
Is it possible for a witness server never submit signatures to claim rewards? The source chain does not know who submitted signatures to destination chain. If this kind of behavior is possible and is a concern, the action of claiming rewards probably should be moved to destination chain.